composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Copyright 2023 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log to `Neptune.ai <https://neptune.ai/>`_."""
+
+__all__ = ['NeptuneLogger']
+
+import os
+import pathlib
+import warnings
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence, Union
+
+import numpy as np
+import torch
+from neptune.internal.utils import verify_type
+
+from composer._version import __version__
+from composer.loggers import LoggerDestination
+from composer.utils import MissingConditionalImportError, dist
+
+if TYPE_CHECKING:
+    from composer import Logger
+    from composer.core import State
+
+
+class NeptuneLogger(LoggerDestination):
+    """LoggerDestination implementation that sends the logs to Neptune.ai.
+
+    Args:
+        project (str, optional): The name of your Neptune project. If nothing is
+            specified, environment variable NEPTUNE_PROJECT will be used.
+        api_token (str, optional): Your Neptune.ai api key. If nothing is specified,
+            environment variable NEPTUNE_API_TOKEN will be used.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            (default: ``True``).
+        log_artifacts (bool, optional): Whether the logger should upload artifacts to Neptune.ai.
```

Comment:
The name `log_artifacts` might be misleading for Neptune users as tracking artifacts is different from uploading them to Neptune.

```suggestion
        upload_artifacts (bool, optional): Whether the logger should upload artifacts to Neptune.ai.
```